components/ws-manager/pkg/manager/create.go

```diff
@@ -582,28 +584,50 @@ func (m *Manager) createDefiniteWorkspacePod(startContext *startWorkspaceContext
 				},
 			}
 
-			// SubPath so that lost+found is not visible
-			pod.Spec.Containers[0].VolumeMounts[0].SubPath = "workspace"
-			// not needed, since it is using dedicated disk
-			pod.Spec.Containers[0].VolumeMounts[0].MountPropagation = nil
+			volumeDevice := corev1.VolumeDevice{
+				Name:       workspaceVolumeName,
+				DevicePath: "/dev/workspace",
+			}
+			pod.Spec.Containers[0].VolumeDevices = append(pod.Spec.Containers[0].VolumeDevices, volumeDevice)
+
+			// get rid of first volume mount as PVC is mounted as block device and will be mounted by workspacekit
+			pod.Spec.Containers[0].VolumeMounts = pod.Spec.Containers[0].VolumeMounts[1:]
 
-			// pavel: 133332 is the Gitpod UID (33333) shifted by 99999. The shift happens inside the workspace container due to the user namespace use.
-			// We set this magical ID to make sure that gitpod user inside the workspace can write into /workspace folder mounted by PVC
-			gitpodGUID := int64(133332)
-			pod.Spec.SecurityContext.FSGroup = &gitpodGUID
+			pod.Spec.Containers[0].VolumeMounts = append(pod.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
+				Name:      "test",
+				MountPath: "/pvc",
+			})
 
-			// add init container to chown workspace subpath, so that it is owned by gitpod user (there is no k8s native way of doing this as of right now)
+			pod.Spec.Volumes = append(pod.Spec.Volumes, corev1.Volume{
+				Name: "test",
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{
+						Medium: corev1.StorageMediumDefault,
+					},
+				},
+			})
+
+			rootUID := int64(0)
+			// add init container to mount PVC block device
 			pod.Spec.InitContainers = append(pod.Spec.InitContainers, corev1.Container{
-				Name:            "chown-workspace",
-				Image:           "busybox",
+				Name:            "mount-workspace-volume",
+				Image:           pod.Spec.Containers[0].Image,
 				ImagePullPolicy: corev1.PullIfNotPresent,
-				Command:         []string{"chown", "133332:133332", "/workspace"},
+				Command:         []string{"/.supervisor/workspacekit", "fsprep"},
+				VolumeDevices:   []corev1.VolumeDevice{volumeDevice},
 				VolumeMounts: []corev1.VolumeMount{
 					{
-						Name:      workspaceVolumeName,
-						SubPath:   "workspace",
-						MountPath: "/workspace",
+						Name:      "test",
+						MountPath: "/pvc",
+					},
+				},
```

Comment:
[This](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#raw-block-volume-support:~:text=Note%3A%20When%20adding%20a%20raw%20block%20device%20for%20a%20Pod%2C%20you%20specify%20the%20device%20path%20in%20the%20container%20instead%20of%20a%20mount%20path.) suggests that within a container, we should specify the device path, instead of the mount path, when mounting a PVC in volumeMode `block`. 

See how in this [example](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#pod-specification-adding-raw-block-device-path-in-container) for a pod spec, there is no volumeMount?

If we do like :point_up: , I assume we'd then have to mount the device from `workspacekit`...and I'm unsure if that's where you're having trouble. :thinking: